src/App.tsx

```diff
@@ -1,50 +1,77 @@
-import React from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
-
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+import { useState } from 'react';
+import { goodsArrayFromServer } from './types/GoodsFromServer';
+import { SortType } from './types/SortType';
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const visibleGoods = [...goodsArrayFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.Alphabetically:
+        return good1.localeCompare(good2);
+      case SortType.Length:
+        return good1.length - good2.length;
+      default:
+        return SortType.Default;
+    }
+  });
+
+  const reset = () => {
+    setSortField('');
+    setIsReverse(false);
+  };
+
+  if (isReverse) {
+    visibleGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortType.Alphabetically)}
+          type="button"
+          className={`button is-info ${sortField === 'alphabetically' ? '' : 'is-light'}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.Length)}
+          type="button"
+          className={`button is-success ${sortField === 'length' ? '' : 'is-light'}`}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setIsReverse(prevReverse => !prevReverse)}
+          type="button"
+          className={`button is-warning ${isReverse ? '' : 'is-light'}`}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(isReverse || sortField) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={reset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map(good => (
+          <li key={good} data-cy="Good">
+            {good}
+          </li>
+        ))}
```

Comment:
According to the checklist, when rendering a list, you must add a `key` prop. However, the key should be a unique identifier for each element. If 'good' is not guaranteed to be unique, consider using a different attribute or a combination that ensures uniqueness.